### ISSUE TITLE
InvalidCastException in ClrPropertySetter.SetClrValue when using nullable computed column with bitwise expression

### ISSUE THREAD

**Issue by mikary** [aspnet/EntityFramework#1262]
When updating nullable computed columns with non-null results of a bitwise operation, ClrPropertySetter.SetClrValue throws an InvalidCastException.

The exception is not thrown when the result is null (not computed).
The exception is not thrown for non-nullable computed columns.

Reproduction
```cs
    [Flags]
    public enum FlagEnum
    {
        None = 0x0,
        AValue = 0x1,
        BValue = 0x2,
    }

    public class EnumItem
    {
        public int EnumItemId { get; set; }

        public FlagEnum FlagEnum { get; set; }

        public FlagEnum? OptionalFlagEnum { get; set; }

        public FlagEnum? CalculatedFlagEnum { get; set; }
    }

    public class NullableContext : DbContext
    {
        public NullableContext(DbContextOptions contextOptions)
            : base(contextOptions)
        {

        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {

            modelBuilder.Entity<EnumItem>()
                .Property(entity => entity.CalculatedFlagEnum)
                .StoreComputed()
                .ForSqlServer().DefaultExpression("FlagEnum | OptionalFlagEnum");
        }

        public DbSet<EnumItem> EnumItems { get; set; }
    }

        [Fact]
        public void Can_use_computed_columns_with_nullable_bitwise_operations()
        {
            var options = new DbContextOptions()
                .UseSqlServer(
                    new SqlConnection(
                        CreateConnectionString("NulableContext")));

            using (var context = new NullableContext(options))
            {
                context.Database.EnsureDeleted();
                context.Database.EnsureCreated();

                context.EnumItems.Add(new EnumItem { FlagEnum = FlagEnum.AValue, OptionalFlagEnum = FlagEnum.BValue });
                context.SaveChanges();
            }
        }
```

**Comment by smitpatel** [comment]
The root cause: Converting boxed int value to nullable enum causes InvalidCastException. [Reference](https://connect.microsoft.com/VisualStudio/feedback/details/593117/cannot-directly-cast-boxed-int-to-nullable-enum)
In above case, the value is computed correctly as 3 but it is boxed since ClrPropertySetter class is generic. When casting this value to TValue which is FlagEnum? here, exception is thrown.
This also explains why it is working for other cases. (non-computed/non-nullable enum)

**Comment by smitpatel** [comment]
fixed via fec52f0b6835559ed3a2eece0ae23e2cd9dd4bc4
